packages/liveblocks-client/src/room.ts

```diff
@@ -1636,45 +1629,51 @@ function prepareAuthEndpoint(
   throw new Error("Internal error. Unexpected authentication type");
 }
 
-function fetchAuthEndpoint(
+async function fetchAuthEndpoint(
   fetch: typeof window.fetch,
   endpoint: string,
   body: {
     room: string;
     publicApiKey?: string;
   }
 ): Promise<any> {
-  return fetch(endpoint, {
+  const res = await fetch(endpoint, {
     method: "POST",
     headers: {
       "Content-Type": "application/json",
     },
     body: JSON.stringify(body),
-  })
-    .then((res) => {
-      if (!res.ok) {
-        throw new AuthenticationError(
-          `Expected a status 200 but got ${res.status} when doing a POST request on "${endpoint}"`
-        );
-      }
+  });
+  if (!res.ok) {
+    throw new AuthenticationError(
+      `Expected a status 200 but got ${res.status} when doing a POST request on "${endpoint}"`
+    );
+  }
 
-      return res.json().catch((er) => {
-        throw new AuthenticationError(
-          `Expected a json when doing a POST request on "${endpoint}". ${er}`
-        );
-      });
-    })
-    .then((authResponse) => {
-      if (typeof authResponse.token !== "string") {
-        throw new AuthenticationError(
-          `Expected a json with a string token when doing a POST request on "${endpoint}", but got ${JSON.stringify(
-            authResponse
-          )}`
-        );
-      }
+  let raw: Json;
+  try {
+    raw = await res.json();
+  } catch (er) {
+    throw new AuthenticationError(
+      `Expected a JSON object when doing a POST request on "${endpoint}". ${er}`
+    );
+  }
 
-      return authResponse;
-    });
+  if (!isJsonObject(raw)) {
+    throw new AuthenticationError(
+      `Expected a JSON object when doing a POST request on "${endpoint}".`
+    );
+  }
```

Comment:
Here is the only small behavioral change I added: it checks to see that the `res.json()` call actually returns a JSON object (not any other JSON value).